torch/csrc/autograd/saved_variable.cpp

```diff
@@ -214,7 +214,10 @@ Variable SavedVariable::unpack(std::shared_ptr<Node> saved_for) const {
 void SavedVariable::set_hooks_and_pack_data(std::unique_ptr<SavedVariableHooks>&& hooks, const Variable& data) {
   hooks_ = std::move(hooks);
   at::NoGradGuard guard;
-  hooks_->call_pack_hook(saved_original_ ? data.tensor_data() : data);
+  const auto version = impl::version_counter(data).current_version();
+  hooks_->call_pack_hook(saved_original_ ? data.detach() : data);
+  TORCH_CHECK(version == impl::version_counter(data).current_version(),
+      "The input to a saved tensor pack hook has been modified by an inplace operation.");
```

Comment:
Can you give a bit more details in this error messasge? At least mention that it is forbidden to modify the given Tensor inplace to avoid un-expected side effect.